src/dotenv/parser.py

```diff
@@ -25,9 +30,20 @@ def make_regex(string, extra_flags=0):
 _double_quote_escapes = make_regex(r"\\[\\'\"abfnrtv]")
 _single_quote_escapes = make_regex(r"\\[\\']")
 
-Binding = NamedTuple("Binding", [("key", Optional[Text]),
-                                 ("value", Optional[Text]),
-                                 ("original", Text)])
+
+try:
+    # this is necessary because we only import these from typing
+    # when we are type checking, and the linter is upset if we
+    # re-import
```

Comment:
What about using `if IS_TYPE_CHECKING:`  instead of trying to re-import?  I think it's OK to use `collections.namedtuple` for all runtime cases (with and without `typing` available), and use `typing.NamedTuple` only for the type-checking case.

Reply:
`IS_TYPE_CHECKING` is only ever true if mypy is running, i.e. it is always false at runtime -- this helps unnecessarily loading the typing imports, so the `try/except` approach allows us to attempt to take advantage of the typed `NamedTuple` if possible.

Using a typed `NameTuple` here explicitly gains you two? benefits I think -- better completion support when creating `Binding` instances in your editor/interactive session, and obviously better type checking -- at the cost of somewhat uglier code with higher maintenance overhead. I don't feel particularly strongly either, only wanted to minimize the functional impact of the changes I was proposing. Let me know which you prefer and I will make the change (in a more timely manner this time around!)

Reply:
OK fair enough.